src/Symfony/Component/Notifier/Bridge/Bluesky/BlueskyTransport.php

```diff
@@ -0,0 +1,227 @@
+                'record' => $post,
+            ],
+        ]);
+
+        try {
+            $statusCode = $response->getStatusCode();
+        } catch (TransportExceptionInterface $e) {
+            throw new TransportException('Could not reach the remote bluesky server.', $response, 0, $e);
+        }
+
+        if (200 === $statusCode) {
+            $content = $response->toArray();
+            $sentMessage = new SentMessage($message, (string) $this);
+            $sentMessage->setMessageId($content['cid']);
+
+            return $sentMessage;
+        }
+
+        try {
+            $content = $response->toArray(false);
+        } catch (DecodingExceptionInterface $e) {
+            throw new TransportException('Unexpected response from bluesky server.', $response, 0, $e);
+        }
+
+        $title = $content['error'] ?? '';
+        $errorDescription = $content['message'] ?? '';
+
+        throw new TransportException(sprintf('Unable to send message to Bluesky: Status code %d (%s) with message "%s".', $statusCode, $title, $errorDescription), $response);
+    }
+
+    private function parseFacets(string $input): array
+    {
+        $facets = [];
+        $text = new ByteString($input);
+
+        // regex based on: https://bluesky.com/specs/handle#handle-identifier-syntax
+        $regex = '#[$|\W](@([a-zA-Z0-9]([a-zA-Z0-9-]{0,61}[a-zA-Z0-9])?\.)+[a-zA-Z]([a-zA-Z0-9-]{0,61}[a-zA-Z0-9])?)#';
+        foreach ($this->getMatchAndPosition($text, $regex) as $match) {
+            $response = $this->client->request('GET', sprintf('https://%s/xrpc/com.atproto.identity.resolveHandle', $this->getEndpoint()), [
+                'query' => [
+                    'handle' => ltrim($match['match'], '@'),
+                ],
+            ]);
+            try {
+                if (200 !== $response->getStatusCode()) {
+                    continue;
+                }
+            } catch (TransportExceptionInterface $e) {
+                $this->logger->error('Could not reach the remote bluesky server. Tried to lookup username.', ['exception' => $e]);
+                throw $e;
+            }
+
+            $did = $response->toArray(false)['did'] ?? null;
+            if (null === $did) {
+                $this->logger->error('Could not get a good response from bluesky server. Tried to lookup username.');
+                continue;
+            }
+
+            $facets[] = [
+                'index' => [
+                    'byteStart' => $match['start'],
+                    'byteEnd' => $match['end'],
+                ],
+                'features' => [
+                    [
+                        '$type' => 'app.bsky.richtext.facet#mention',
+                        'did' => $did,
+                    ],
+                ],
+            ];
+        }
+
+        // partial/naive URL regex based on: https://stackoverflow.com/a/3809435
+        // tweaked to disallow some training punctuation
```

Comment:
```suggestion
        // tweaked to disallow some trailing punctuation
```
?

Reply:
Sure. This comment was copied from the official python example.